docs/standard/garbage-collection/large-object-heap.md

```diff
@@ -0,0 +1,313 @@
+
+User code can only allocate in generation 0 (small objects) or the LOH (large objects). Only the GC can “allocate” objects in generation 1 (by promoting survivors from generation 0) and generation 2 (by promoting survivors from generations 1 and 2).
+
+When a garbage collection is triggered, the GC traces through the live objects and compacts them. But because compaction is expensive, the GC *sweeps* the LOH; it makes a free list out of dead objects that can be reused later to satisfy large object allocation requests. Adjacent dead objects are made into one free object.
+
+The .NET Framework 4.5.1 added an option for users to specify that the LOH should be compacted during the next full blocking GC. And in the future, .NET may decide to compact the LOH automatically. This means that, if you allocate large objects and want to make sure that they don’t move, you should still pin them.
+
+Figure 1 illustrates a scenario where the GC forms generation 1 after the first generation 0 GC where `Obj1` and `Obj3` are dead, and it forms generation 2 after the first generation 1 GC where `Obj2` and `Obj5` are dead. Note that this and the following figures are only for illustration purposes; they contain very few objects to better show what happens on the heap. In reality, many more objects are typically involved in a GC.
+
+![Figure 1: A gen 0 GC and a gen 1 GC](media/loh/loh-figure-1.jpg)   
+Figure 1: A generation 0 and a generation 1 GC.
+
+Figure 2 shows that after a generation 2 GC which saw that `Obj1` and `Obj2` are dead, the GC forms contiguous free space out of memory that used to be occupied by `Obj1` and `Obj2`, which then was used to satisfy an allocation request for `Obj4`. The space after the last object, `Obj3`, to end of the segment can also be used to satisfy allocation requests.
+ 
+![Figure 2: After a gen 2 GC](media/loh/loh-figure-2.jpg)  
+Figure 2: After a generation 2 GC
+
+If there isn't enough free space to accommodate the large object allocation requests, the GC first attempts to acquire more segments from the OS. If that fails, it triggers a generation 2 GC in the hope of freeing up some space.
+
+During a generation 1 or generation 2 GC, the garbage collector releases segments that have no live objects on them back to the OS by calling the [VirtualFree function](https://msdn.microsoft.com/library/windows/desktop/aa366892(v=vs.85).aspx). Space after the last live object to the end of the segment is decommitted. And the free spaces remain committed though they are reset, meaning that the OS doesn’t need to write data in them back to disk. 
```

Comment:
I'd like to add a clarification here (sorry I didn't catch it when I sent you the doc). 
> Space after the last live object to the end of the segment is decommitted

should be

>  Space after the last live object to the end of the segment is decommitted (except on the ephemeral segment where gen0/1 live, we do keep some committed because you will be allocating in it right away).